projects/plugins/jetpack/extensions/store/wordpress-com/reducer.ts

```diff
@@ -97,16 +97,35 @@ export default function reducer( state = INITIAL_STATE, action ) {
 		}
 
 		case ACTION_INCREASE_AI_ASSISTANT_REQUESTS_COUNT: {
-			// Increase request count;
 			const requestsCount = state.features.aiAssistant.requestsCount + action.count;
 
+			const currentTierValue = state.features.aiAssistant.currentTier?.value;
+
+			const isFreeTierPlan =
+				( typeof currentTierValue === 'undefined' && ! state.features.aiAssistant.hasFeature ) ||
+				currentTierValue === 1;
+
+			const isUnlimitedTierPlan =
+				( typeof currentTierValue === 'undefined' && state.features.aiAssistant.hasFeature ) ||
+				currentTierValue === 1;
+
+			const requestsLimit =
+				isUnlimitedTierPlan || isFreeTierPlan // @todo: update once tier data is available
+					? state.features.aiAssistant.requestsLimit
+					: state.features.aiAssistant.currentTier?.limit;
+
+			const currentCount =
+				isUnlimitedTierPlan || isFreeTierPlan // @todo: update once tier data is available
+					? requestsCount
+					: state.features.aiAssistant.currentTier?.limit;
+
 			/**
 			 * Compute the AI Assistant Feature data optimistically,
 			 * based on the Jetpack_AI_Helper::get_ai_assistance_feature() helper.
 			 *
 			 * @see _inc/lib/class-jetpack-ai-helper.php
 			 */
-			const isOverLimit = requestsCount >= state.features.aiAssistant.requestsLimit;
+			const isOverLimit = currentCount >= requestsLimit;
 			const requireUpgrade = isOverLimit && ! state.features.aiAssistant.hasFeature;
```

Comment:
I know this line is already deployed, but it will fail when the user has a tier. When they have a tier, the `hasFeature` will be `true`, so the condition will always fail.

Maybe we shoud just check if it `isOverLimit`. When the site is over the limit, they require an upgrade.

Reply:
> When they have a tier, the hasFeature will be true

This is a mistake from my PoV. `hasFeature` should reflect whether the site supports the feature, no matter the reasons.

Reply:
🙇‍♂️ Addressed https://github.com/Automattic/jetpack/pull/34083/commits/260106b6c55e711adb78971b0a956e452f0ebde7